devolutions-gateway/src/api/fwd.rs

```diff
@@ -237,3 +254,288 @@ where
         }
     }
 }
+
+async fn fwd_http(
+    State(state): State<DgwState>,
+    BridgeToken(claims): BridgeToken,
+    extract::Path(session_id): extract::Path<Uuid>,
+    ConnectInfo(source_addr): ConnectInfo<SocketAddr>,
+    mut request: axum::http::Request<axum::body::Body>,
+) -> Result<Response, HttpError> {
+    use axum::extract::FromRequestParts as _; // from_request_parts
+    use axum::http::header;
+    use axum::http::Response;
+    use core::str::FromStr;
+    use http_body_util::BodyExt as _; // into_data_stream
+    use std::sync::LazyLock;
+    use tokio_rustls::rustls;
+    use tokio_tungstenite::connect_async_tls_with_config;
+
+    // Default HTTP client for typical usage.
+    static CLIENT: LazyLock<reqwest::Client> = LazyLock::new(reqwest::Client::new);
+
+    // Dangerous HTTP client, only to be used when absolutely necessary.
+    // E.g.: VMware services are often using untrusted self-signed certificates.
+    static DANGEROUS_CLIENT: LazyLock<reqwest::Client> = LazyLock::new(|| {
+        reqwest::Client::builder()
+            .danger_accept_invalid_hostnames(true)
+            .danger_accept_invalid_certs(true)
+            .build()
+            .expect("parameters known to be valid only")
+    });
+
+    static DANGEROUS_TLS_CONNECTOR: LazyLock<tokio_tungstenite::Connector> = LazyLock::new(|| {
+        rustls::client::ClientConfig::builder()
+            .dangerous()
+            .with_custom_certificate_verifier(Arc::new(crate::tls::danger::NoCertificateVerification))
+            .with_no_client_auth()
+            .pipe(Arc::new)
+            .pipe(tokio_tungstenite::Connector::Rustls)
+    });
+
+    const REQUEST_TARGET_PARAM: Parameter<String> =
+        Parameter::new("Dgw-Request-Target", |params| params.request_target.take());
+
+    const DANGEROUS_TLS_PARAM: Parameter<bool> =
+        Parameter::new("Dgw-Dangerous-Tls", |params| params.dangerous_tls.take());
+
+    #[derive(Deserialize)]
+    #[serde(rename_all = "camelCase")]
+    struct QueryParams {
+        request_target: Option<String>,
+        dangerous_tls: Option<bool>, // QUESTION: Maybe we should bake this parameter into the Bridge token.
```

Comment:
I may not have enough context to answer this. Is this blocking for merge, could it me resolved in follow-up?

It first glance, it indeed looks not safe to me to select a dangerous TLS connector via HTTP param, adding it to signed token claims sounds about right

Reply:
Not a blocker, and the API is behind the unstable flag anyway. It's not as bad as it may look since we still need a token to use this endpoint, and we are just disabling checks usually done on client side, usually disabled at the client'a discretion. Devolutions Gateway itself is not impacted by that. Putting this in the token may make things more complicated than they really need to be. We would need DVLS to know when an unsafe TLS token should be provided or not. This may end up being put in the query params of the DVLS endpoint so the client can chooses anyway. At this point it's just a less flexible solution (we can't mix normal and unsafe calls with a single token), ultimately potentially less secure than it needs to be. What do you think?

Reply:
> It's not as bad as it may look since we still need a token to use this endpoint, and we are just disabling checks usually done on client side

In that case, putting it in params sounds fine to me 👍